### PR TITLE
refactor: Use URL.hostname

### LINKED ISSUE #1007: Bug: Sanitize URL for the repo form
### Describe the bug

https://github.com/open-sauced/open-sauced/blob/d7f554e68064e4079e39b7e01ce1d5a6ce130517/src/lib/util.js#L40-L40

This code allows `github.com` URLs but it could be anywhere. To make this secure, we should probably switch to regex. 

[CodeQL](https://securitylab.github.com/tools/codeql/) also alerted us of this.
### To Reproduce

1. Head to opensauced.pizza and log in.
2. Add a URL that includes github.com but not a repo

### Screenshots

Adding a screenshot from CodeQL

Screen Shot 2021-05-19 at 5 24 22 AM

### PR DESCRIPTION
**Pull request by mtfoley**
## What type of PR is this? (check all applicable)

- [x] ♻️ Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 👷 Optimization
- [ ] 📝 Documentation Update
- [ ] 🔖 Release
- [ ] 🚩 Other

## Description
Fixes #1007 by checking URL.hostname.  I tried to refactor this file down to a single function and the test suite convinced me I was being too clever.

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 readme
- [ ] 📜 contributing.md
- [ ] 📓 docs
- [x] 🙅 no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?
hackerman

### CODE REVIEW

**Comment by 0-vortex** [comment]
> When you say right to left I assume you mean split by "/" and grab the last two non-blank elements? Could work, but the test suite covers a bunch of cases and I think one of them is failing on the wrong host name and also on misspelled URL's. An alternate might be a regex like /(https*://github.com/)*/([^/]+)/([^/]+)/*/ and grabbing some of the match groups. Let's see how composing an untested regex in a comment works out for me...

Ah I can tell by your regex you are already thinking about supporting lookbehind for potential multi-platform checking (if onegraph for example lets us add gitlab repos) but I was thinking more simplistic in the sense of parsing strings with `([^\/]*)\/([^\/]*)$`:

```shell
# https://regex101.com
nodejs/node
https://github.com/vercel/next.js
https://any-github-proxy.com/vercel/next.js
https://githhbb.com/vercel/next.js
https://i.can.t.spell.for.the.life.of.me.com/vercel/next.js
```

The regex can be further simplified but this only works if we extract owner/repo name from the end of the string and create a `new URL(owner/repo, 'github.com')` essentially forcing the parameter to always create the url with github hostname in order to collapse the 3 function logic into just the regex and to remove the codeql concern by creating the URL object right after the match. 

Feel like correcting user mistakes in the process rather than throwing multiple types of errors is more beneficial in actions connected to the onboarding process :D

File: src/lib/util.js [hunk 1]
```diff
@@ -37,9 +37,9 @@ function absoluteUrlValidator(url) {
 
     const [owner, repo] = relativeRepoUrl.split("/");
 
-    if (!url.toLowerCase().includes("github.com"))
+    if (urlObject.hostname.toLowerCase() !== "github.com") {
       return [false, null];
```
Comment (filiptronicek) [inline src/lib/util.js:41]: This looks almost right, I would just remove the `.toLowerCase()` method, because the URL constructor does it for us.

Comment (mtfoley) [reply src/lib/util.js:41]: Could have sworn it failed the test suite. I'll change it and make sure.

**Comment by 0-vortex** [comment]
> @0-vortex I’d say the changes you’re talking about could just go into a different PR for refactor. I def see value in refactoring to one function (and I tried to) but I did not have the bandwidth at the time to square up to that test suite.

Completely agree, will open up an enhancement issue if everyone agrees

**Review by filiptronicek — APPROVED** [review]
Works just fine from my quick testing, LGTM

**Review by bdougie — APPROVED** [review]
LGTM

**Pull request merged**